tests/browser/test/toolbox_drag_test.js

```diff
@@ -0,0 +1,202 @@
+
+    const vertInView =
+      rect.top >= blocklyRect.top && rect.bottom <= blocklyRect.bottom;
+    const horInView =
+      rect.left >= blocklyRect.left && rect.right <= blocklyRect.right;
+
+    return vertInView && horInView;
+  }, element);
+}
+
+/**
+ * Get how many top-level blocks there are in the specified category.
+ * @param browser The active WebdriverIO Browser object.
+ * @param categoryName The name of the category to inspect.
+ * @returns A Promise resolving to the number of top-level blocks in the specified
+ *     category's flyout.
+ */
+async function getBlockCount(browser, categoryName) {
+  const category = await getCategory(browser, categoryName);
+  await category.click();
+  await browser.pause(PAUSE_TIME);
+
+  const blockCount = await browser.execute(() => {
+    return Blockly.getMainWorkspace()
+      .getFlyout()
+      .getWorkspace()
+      .getTopBlocks(false).length;
+  });
+
+  // Unicode escape to close flyout.
+  await browser.keys(['\uE00C']);
+  await browser.pause(PAUSE_TIME);
+  return blockCount;
+}
+
+async function isBlockDisabled(browser, i) {
+  return await browser.execute((n) => {
+    return !Blockly.getMainWorkspace()
+      .getFlyout()
+      .getWorkspace()
+      .getTopBlocks()
+      [n].isEnabled();
+  }, i);
+}
+
+/**
+ * Loop over a list of categories and click on each one to open it.
+ * @param browser The WebdriverIO Browser instance for this test.
+ * @param categoryList An array of category names, as strings.
+ * @returns A Promise that resolves when all actions have finished.
+ */
+async function openCategories(browser, categoryList) {
+  let failureCount = 0;
+  for (const categoryName of categoryList) {
+    const blockCount = await getBlockCount(browser, categoryName);
+
+    try {
+      for (let i = 0; i < blockCount; i++) {
+        const category = await getCategory(browser, categoryName);
+        await category.click();
+        if (await isBlockDisabled(browser, i)) {
+          // Unicode escape to close flyout.
+          await browser.keys(['\uE00C']);
+          await browser.pause(PAUSE_TIME);
+        } else {
+          let flyoutBlock = await browser.$(
+            `.blocklyFlyout .blocklyBlockCanvas > g:nth-child(${3 + i * 2})`,
+          );
+          if (!(await elementInBounds(browser, flyoutBlock))) {
+            await scrollFlyout(browser, 0, 500);
+          }
+
+          await flyoutBlock.dragAndDrop({x: 50, y: 0});
+          await browser.pause(PAUSE_TIME);
+          // Should be one top level block on the workspace.
+          const topBlockCount = await browser.execute(() => {
+            return Blockly.getMainWorkspace().getTopBlocks(false).length;
+          });
+
+          if (topBlockCount != 1) {
+            failureCount++;
+            console.log(`fail: block ${i} in category ${categoryName}`);
+          }
+
+          // Clean up between blocks so they can't interact with each other
```

Comment:
Nit: End comment with a period.

(I know there are other places in this PR / the tests that don't conform. But for new stuff might as well)

Reply:
Done!